// react-redux-form-materialize/lib/controls/fileInput.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends2 = require('babel-runtime/helpers/extends');

var _extends3 = _interopRequireDefault(_extends2);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _omit = require('lodash/omit');

var _omit2 = _interopRequireDefault(_omit);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

class FileInput extends _react2.default.Component {
  constructor(...args) {
    var _temp;

    return _temp = super(...args), this.onFileClick = () => this.props.onChange(null), _temp;
  }

  render() {
    const { props } = this;

    const inputValue = props.innerState.value && props.innerState.value.length ? props.innerState.value.reduce((prevVal, curVal) => prevVal.concat(curVal.name, ', '), []) : [];

    if (inputValue.length) {
      inputValue.pop();
    }

    const fieldClass = (0, _classnames2.default)('file-field', 'input-field', props.className);

    const btnClassName = (0, _classnames2.default)('btn', {
      disabled: props.disabled
    });

    const textInputClassName = (0, _classnames2.default)('file-path', 'validate', {
      [props.textClassName]: props.textClassName
    });

    const inputProps = (0, _omit2.default)(props, ['placeholder', 'innerState', 'textClassName', 'messages', 'buttonText', 'onFileClick', 'className']);

    return _react2.default.createElement(
      'div',
      { className: fieldClass },
      _react2.default.createElement(
        'div',
        { className: btnClassName },
        _react2.default.createElement(
          'span',
          null,
          props.buttonText
        ),
        _react2.default.createElement('input', (0, _extends3.default)({}, inputProps, { onClick: this.onFileClick, type: 'file' }))
      ),
      _react2.default.createElement(
        'div',
        { className: 'file-path-wrapper' },
        _react2.default.createElement('input', { disabled: props.disabled, value: inputValue.join(''), className: textInputClassName, type: 'text',
          readOnly: true, placeholder: props.placeholder })
      )
    );
  }
}
exports.default = FileInput;
FileInput.propTypes = {
  innerState: _propTypes2.default.object.isRequired
};
